packages/backend/src/apps/m365-excel/common/get-eligible-tenant-keys.ts

```diff
@@ -0,0 +1,34 @@
+import appConfig from '@/config/app'
+import { type M365TenantKey, m365TenantKeys } from '@/config/app-env-vars/m365'
+import { getLdFlagValue } from '@/helpers/launch-darkly'
+
+export default async function getEligibleTenantKeys(
+  userEmail: string,
+): Promise<ReadonlySet<M365TenantKey>> {
+  const result = new Set<M365TenantKey>()
+
+  // Get eligible tenants from LD.
+  const potentialTenants = await Promise.all(
+    m365TenantKeys.map(async (tenantKey) => ({
+      key: tenantKey,
+      isEligible: await getLdFlagValue(
+        `backend_m365_tenant_${tenantKey}`,
+        userEmail,
+        false,
+      ),
```

Comment:
this doesn't gate those who already previously had access to these tenants. is this intended?

Reply:
Yep. 

> To prevent confusion, we still allow users to see old M365 connections, even if they have become ineligible for it (e.g. we blocked them on LD). They will instead get an error message if they select that connection, but have become ineligible (implemented in a later PR).